src/BridgedRoom.ts

```diff
@@ -251,16 +255,27 @@ export class BridgedRoom {
                 emojiKeyName = emojiKeyName.substring(1, emojiKeyName.length - 1);
             }
         }
+        let client: WebClient = this.botClient;
+        let id: string|undefined;
+        const puppet = await this.main.clientFactory.getClientForUserWithId(this.SlackTeamId!, message.sender);
+        if (puppet) {
+            client = puppet.client;
+            id = puppet.id;
+        }
 
-        // TODO: This only works once from matrix as we are sending the event as the
+        // TODO: This only works once from matrix if we are sending the event as the
         // bot user.
-        const res = await this.botClient.reactions.add({
+        const res = await client.reactions.add({
             as_user: false,
             channel: this.slackChannelId,
             name: emojiKeyName,
             timestamp: event.slackTs,
         });
 
+        if (id) {
```

Comment:
Don't we also want to put the event ids of reactions into the db for the bot?

Reply:
The bot's messages are automatically filtered out